### PR TITLE
experiment spark

### LINKED ISSUE #178: Create a Dockerfile for hellbender-protected
It should install HDF5 and include instructions for running gradle test.

### LINKED ISSUE #182: Use MLLib for SVD in create PoN

### PR DESCRIPTION
**Pull request by LeeTL1220**
Closes #182 
Closes #178

### CODE REVIEW

**Comment by LeeTL1220** [comment]
Travis issues with ``continuous-integration/travis-ci/push`` please ignore failure there.

**Comment by LeeTL1220** [comment]
@samuelklee All unit tests pass on laptop.  Can you review while we look into Travis CI issues?

**Comment by samuelklee** [comment]
back to @LeeTL1220.  apologies for the very nitpicky review---almost all of my comments concern superficial matters of style, but there are one or two of substance...

also, i'm guessing you will probably want to squash and change the commit message before merging.

File: src/main/java/org/broadinstitute/hellbender/utils/SparkToggleCommandLineProgram.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        JavaSparkContext ctx = null;
+        if (!isDisableSpark) {
+//            ctx = SparkContextFactory.getSparkContext(getProgramName(), sparkMaster);
+//            ctx.getConf().set("spark.driver.maxResultSize", "0")
+//                    .set("spark.driver.userClassPathFirst", "true")
+//                    .set("spark.executor.userClassPathFirst", "true")
+//                    .set("spark.io.compression.codec", "lzf")
+//                    .set("spark.yarn.executor.memoryOverhead", "600");
+
+            SparkConf sparkConf = new SparkConf().setAppName(getProgramName())
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/utils/SparkToggleCommandLineProgram.java]: I've patched hellbender `SparkContextFactory.getSparkContext` to fix this problem.  Update to https://oss.sonatype.org/content/repositories/snapshots/org/broadinstitute/hellbender/GATK.4.alpha-628-g94107b8-SNAPSHOT/

**Comment by LeeTL1220** [comment]
@samuelklee Back at ya

**Comment by samuelklee** [comment]
a few more minor things, but otherwise good to merge (after squashing, etc)!

**Comment by LeeTL1220** [comment]
@samuelklee Tests failed due to out of memory.  Passes locally.

**Comment by samuelklee** [comment]
looks good to me, good to merge after rebasing/squashing/renaming the commit.  maybe check if @lbergelson's latest commit fixes the OOM failures?

**Comment by LeeTL1220** [comment]
I already did the rebase and it did not fix.

**Pull request merged**